qutebrowser/config/configtypes.py

```diff
@@ -798,6 +799,17 @@ class File(BaseType):
 
     typestr = 'file'
 
+    def transform(self, value):
+        if not value:
+            return None
+        value = os.path.expanduser(value)
+        if not os.path.isabs(value):
+            if standarddir.config():
```

Comment:
I'd move this check to `validate` - right now I think your code will raise an exception when started with `-c ''` (no configuration), because in `validate` you don't check for `None`. And when you check for there, there's no need to check in `transform` as well.

Also, I'd be glad if you could add a test for that scenario, though if you get stuck there let me know and I'll add it. It would probably look something like this (untested):

```python
def test_validate_rel_config_none(self, os_path, monkeypatch):
    """Test with a relative path and standarddir.config() returning None."""
    monkeypatch.setattr('qutebrowser.config.configtypes.standarddir.config', lambda: None)
    with pytest.raises(configexc.ValidationError):
        self.t.validate('foobar')
```